canister/src/validation.rs

```diff
@@ -76,12 +76,21 @@ mod test {
         #![proptest_config(ProptestConfig::with_cases(10))]
         #[test]
         fn validation_context(
-            stability_threshold in 1..150u32,
+            stability_threshold_start in 1..150u32,
             num_blocks in 1..250u32,
         ) {
             let num_transactions_in_block = 1;
             let network = Network::Regtest;
-            let blocks = build_chain(network, num_blocks, num_transactions_in_block);
+
+            let stability_threshold = stability_threshold_start + 1;
+
+            let blocks_start = build_chain(network, num_blocks, num_transactions_in_block);
+
+            let mut blocks = vec![];
+
+            for block in blocks_start.into_iter(){
+                blocks.push(block.with_mock_dificulty(1));
+            }
```

Comment:
I think you can do this (untested), which is a bit more "Rustic":

```suggestion
            let blocks: Vec<_> = build_chain(network, num_blocks, num_transactions_in_block)
              .into_iter().map(|b| b.with_mock_difficulty(1)).collect();
```